Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -361,11 +369,41 @@ static size_t computeRangeSize(uint64_t size, size_t offset)
     return m_emptyBuffer.size() ?: m_size;
 }
 
+bool Buffer::isValid() const
+{
+    return isDestroyed() || m_buffer;
+}
+
 bool Buffer::isDestroyed() const
 {
     return state() == State::Destroyed;
 }
 
+void Buffer::recomputeMaxIndexValues() const
+{
+    if (!(m_usage & WGPUBufferUsage_Index))
+        return;
+
+    NSUInteger lengthInBytes = m_buffer.length;
+    auto bufferPtr = static_cast<uint8_t*>(m_buffer.contents);
+    RELEASE_ASSERT(bufferPtr);
+    m_max16BitIndex = 0;
+    m_max32BitIndex = 0;
+    uint8_t* bufferEnd = bufferPtr + lengthInBytes;
+    for (; (bufferPtr += sizeof(uint32_t)) <= bufferEnd; bufferPtr += sizeof(uint32_t)) {
+        m_max32BitIndex = std::max(*reinterpret_cast<uint32_t*>(bufferPtr), m_max32BitIndex);
```

Comment:
is `bufferPtr` guaranteed to be correctly aligned to avoid misaligned loads? (Maybe that's not an issue on AS which is think behaves like x86 to make porting easier)

Reply:
bufferPtr is the contents pointer of a MTLBuffer instance, so this is guaranteed to be 16 byte aligned